### ISSUE TITLE
Feature Request: Ordering of courses by the result of specific questions

### ISSUE THREAD

**Issue by jchromik** [fsr-itse/EvaP#806]
When trying to find out which courses I would like to attend during the next semester, I am more interested in the results of specific questions than in the overall grade of a course. For example, if I already know that I will have little time during the next semester, I am mostly interested in the results of the question "I think the expenditure of time for the project seminar was appropriate.".

I would like to have a feature that allows ordering the courses by the results of a course to specific questions.
This could be implemented by parsing the text in the search bar for keywords like `orderby:`. In my example I would write `orderby: time_exp` (or something similar) whereas `time_exp` is the short form of / a symbol for the question "I think the expenditure of time for the project seminar was appropriate.".

**Comment by karyon** [comment]
searching for courses with the lowest time expenditure is something we don't want to support. 

while in general the results when sorting by arbitrary questions might be interesting, this alone doesn't justify the complexity of the implementation. you see other use cases for this feature?

**Comment by jchromik** [comment]
As far as I can see, most questions are interesting sorting keys, since the question would not be asked if they were not interesting. For example "Necessary preliminary knowledge was adequately specified in advance." could be important for master students from other universities due to the fact that they may not have the same preliminary knowledge as a master student who attended the belonging bachelor's track before.
Also, depending on what a very student wants to achieve with his or her studies, "I learned something in the lecture.", "The techniques taught in the lecture seemed to be up to date.", "The lecture is important for my studies.", or "I enjoyed the lecture." are question that are valuable sorting criteria.

Using the same syntax, other keys could be parsed from the search bar, e.g. for filtering by average grade, number of voters, type, or responsible lecturer.
